File: fix_translations.py
import json
import re

with open('translations.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract JSON part
prefix = "const translations = "
start_idx = content.find(prefix) + len(prefix)
end_idx = content.rfind(";")
json_str = content[start_idx:end_idx]

data = json.loads(json_str)

new_data = {}
for lang, trans in data.items():
    new_data[lang] = {}
    for key, value in trans.items():
        # Trim leading/trailing whitespace and normalize internal spaces
        new_key = re.sub(r'\s+', ' ', key.strip())
        new_value = re.sub(r'\s+', ' ', value.strip())
        new_data[lang][new_key] = new_value

# Write back
new_content = prefix + json.dumps(new_data, indent=2, ensure_ascii=False) + ";\n"

with open('translations.js', 'w', encoding='utf-8') as f:
    f.write(new_content)

print("translations.js updated successfully via python!")

File: fix_translations.js
const fs = require('fs');

let content = fs.readFileSync('translations.js', 'utf8');

// A simple way to load the translations object
const scriptContent = content + '\nmodule.exports = translations;';
fs.writeFileSync('temp_translations.js', scriptContent);
const translations = require('./temp_translations.js');

const newTranslations = {};

for (const lang in translations) {
    newTranslations[lang] = {};
    for (const key in translations[lang]) {
        // The script matching does: match[2].replace(/\s+/g, ' ')
        // So the key should be exactly that.
        // It's the original trimmed text with internal spaces normalized.
        const newKey = key.trim().replace(/\s+/g, ' ');
        const newValue = translations[lang][key].trim().replace(/\s+/g, ' ');
        newTranslations[lang][newKey] = newValue;
    }
}

const newContent = 'const translations = ' + JSON.stringify(newTranslations, null, 2) + ';\n';
fs.writeFileSync('translations.js', newContent);
console.log("Successfully fixed translations.js");
